path_login.go

```diff
@@ -78,8 +81,9 @@ func (b *backend) pathLoginUpdate(ctx context.Context, req *logical.Request, dat
 	authenticateRequestHeaders := requestHeaders.(http.Header)
 
 	// Find the targetUrl and Method
-	finalLoginPath := PathVersionBase + fmt.Sprintf(PathBaseFormat, roleName)
-	method, targetUrl, err := requestTargetToMethodURL(authenticateRequestHeaders[HdrRequestTarget], PathLoginMethod, finalLoginPath)
+	finalLoginPathRegex := fmt.Sprintf("^%s%s$", PathVersionBase,
```

Comment:
Though the use of a dynamic regex here looks OK (since `roleName` is going to be constrained), it might be more complex and difficult to review than necessary. Can `requestTargetToMethodURL` use `strings.Split(s, "/")` and explicity check the parts? That would remove the regex and probably(?) improve readability.

Reply:
Yep, I agree that using `strings.Split(s, "/")` and checking the parts would be better than using a regex here.

I've updated this PR to use that in 4b0bd35. Let me know your thoughts.